lib/matplotlib/axes/_base.py

```diff
@@ -551,6 +551,9 @@ class _AxesBase(martist.Artist):
 
     _subclass_uses_cla = False
 
+    dataLim: mtransforms.Bbox
+    """The bounding `.Bbox` enclosing all data displayed in the Axes."""
```

Comment:
We need a place to put the docstring. IMHO a class-level annotation is the best location (though we normally don't write types in the py files).